### PR TITLE
Explanation on aware interfaces

### PR DESCRIPTION
**Pull request by hannesvdvreken**
Subsplit of what I wanted to explain in the abstract-classes-are-an-antipattern blog post.

@madewithlove/development

### CODE REVIEW

File: _posts/2015-09-25-aware-interfaces.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+```
+Zend/Db/Adapter/AdapterAwareInterface.php
+Zend/EventManager/EventManagerAwareInterface.php
+Zend/EventManager/SharedEventAggregateAwareInterface.php
+Zend/Log/LoggerAwareInterface.php
+```
+
+These packages often come with a reusable trait that gives you pluggable implementations for these interfaces out of the box. For previous examples there are the [`EmitterAwareTrait`](https://github.com/thephpleague/event/blob/master/src/EmitterAwareTrait.php) and the [`ImmutableContainerAwareTrait`](https://github.com/thephpleague/container/blob/master/src/ImmutableContainerAwareTrait.php).
+
+I find these interfaces useful for setting up an IoC container to automatically "setter-inject" a dependency. With Laravel 5.0+'s container one configure this:
```
Comment (tonysm) [inline _posts/2015-09-25-aware-interfaces.md]: `With Laravel 5.0+'s container one configure this` maybe rephrase this?

**Comment by tonysm** [comment]
:+1:

File: _posts/2015-09-25-aware-interfaces.md (hunk 1 above)
Comment (hannesvdvreken) [inline _posts/2015-09-25-aware-interfaces.md]: Ah yeah, that sentence is broken :p

**Comment by hannesvdvreken** [comment]
Thanks for reviewing @tonysm
